include/sparrow/buffer/dynamic_bitset.hpp

```diff
@@ -561,24 +647,156 @@ namespace sparrow
         zero_unused_bits();
     }
 
+    template <random_access_range B>
+    constexpr void dynamic_bitset_base<B>::clear() noexcept
+    {
+        resize(0);
+    }
+
+    template <random_access_range B>
+    constexpr dynamic_bitset_base<B>::iterator
+    dynamic_bitset_base<B>::insert(const_iterator pos, value_type value)
+    {
+        return insert(pos, 1, value);
+    }
+
+    template <random_access_range B>
+    constexpr dynamic_bitset_base<B>::iterator
+    dynamic_bitset_base<B>::insert(const_iterator pos, size_type count, value_type value)
+    {
+        SPARROW_ASSERT_TRUE(cbegin() <= pos);
+        SPARROW_ASSERT_TRUE(pos <= cend());
+        const auto index = static_cast<size_type>(std::distance(cbegin(), pos));
+        const size_type old_size = size();
+        const size_type new_size = old_size + count;
+
+        // TODO: The current implementation is not efficient. It can be improved.
+
+        resize(new_size);
+
+        for (size_type i = old_size + count - 1; i >= index + count; --i)
+        {
+            set(i, test(i - count));
+        }
+
+        for (size_type i = 0; i < count; ++i)
+        {
+            set(index + i, value);
+        }
+
+        auto block = data() + block_index(index);
+        return iterator(this, block, bit_index(index));
+    }
+
+    template <random_access_range B>
+    template <std::input_iterator InputIt>
+    constexpr dynamic_bitset_base<B>::iterator
+    dynamic_bitset_base<B>::insert(const_iterator pos, InputIt first, InputIt last)
+    {
+        SPARROW_ASSERT_TRUE(cbegin() <= pos);
+        SPARROW_ASSERT_TRUE(pos <= cend());
+        const auto index = static_cast<size_type>(std::distance(cbegin(), pos));
+        const size_type old_size = size();
+        const size_type count = static_cast<size_type>(std::distance(first, last));
+        const size_type new_size = old_size + count;
+
+        resize(new_size);
+
+        // TODO: The current implementation is not efficient. It can be improved.
```

Comment:
Consider creating an issue about this type's performance once this is merged?

Reply:
Ok added: #201